#1647971408 integrate mocha and chai

igbominadeveloper opened the pull request:
#### What does this PR do?
Sets up Unit testing using mocha and chai libraries

#### Description of Task to be completed?
- Install Mocha
- Install chai-http
- Install chai assertion library

#### How should this be manually tested?
- clone the repo.
- run npm install
- checkout the ```ch-setup-unit-testing-164797140``` branch
- add any test cases to the articles.spec.js file
- save the file and run ```npm test``` in your terminal

#### Any background context you want to provide?
- None

#### What are the relevant pivotal tracker stories?(if applicable)
#164797140

#### Screenshots
- None

Commit:
[chore #164797141] create coveralls integration

igbominadeveloper committed:
[chore #1647971408] integrate mocha and chai

igbominadeveloper committed:
[chore #1647971408] integrate mocha and chai

Commit:
[fix 164797141]fix uninstalled nyc package

Commit:
[fix 164797141]add coveralls badge to readme

Japhethca reviewed (changes requested):
please remove cross-env package

Japhethca commented on .babelrc:
```diff
@@ -0,0 +1,6 @@
+    "plugins": [
+        ["@babel/transform-runtime"]
+    ]
+}
```
new line here

igbominadeveloper replied:
I have added it to it

emmsdan committed:
#164797137 Chore Configured eslint (#2)

* chore(eslint): Configured eslint and editorconfig
  - standard used: airbnb
  - allows uniform configuration across developers editor

eben-k committed:
Merge branch 'develop' into ch-integrate-coveralls-164797141

eben-k committed:
Merge pull request #3 from andela/ch-integrate-coveralls-164797141

#164797141 Create coveralls integration

igbominadeveloper committed:
[chore] add space to endof line in babelrc file

igbominadeveloper committed:
[chore] add space to endof line in babelrc file

igbominadeveloper committed:
[chore] clean up package.json file

- remove cross-env from test script
- remove unused packages

igbominadeveloper committed:
[chore] clean up package.json file

- remove cross-env from test script
- remove unused packages

igbominadeveloper committed:
[chore] rename test file to comply with agreed convention

igbominadeveloper committed:
[chore] rename test file to comply with agreed convention

NedyUdombat committed:
Add Heroku generated app.json

igbominadeveloper committed:
Delete package-lock.json

igbominadeveloper committed:
[chore] rebase with develop
